hintparser.y

```diff
@@ -240,6 +241,14 @@ TableOptimizerHintOpt:
 			HintData: $4,
 		}
 	}
+|	"NTH_PLAN" '(' QueryBlockOpt hintIntLit ')'
+	{
+		$$ = &ast.TableOptimizerHint{
+			HintName: model.NewCIStr($1),
+			QBName:   model.NewCIStr($3),
+			HintData: int64($4),
```

Comment:
what happens with `NTH_PLAN(18446744073709551615)` ?

Reply:
@kennytm  The planner will check the number `x` given by the hint. An error will be given if `x<-1` or `x>100`.